### PR TITLE
Make pipeline discovery more flexible

### PR DESCRIPTION
**Pull request by jitu5**
Resolves https://github.com/kedro-org/vscode-kedro/issues/18

This PR improves the Kedro extension's ability to detect and navigate to pipeline files by supporting more flexible file naming conventions.

- Updated the references function to use glob with recursive=True, allowing it to find all files containing "pipeline" in their names, even in nested subdirectories.
- Replaced the fixed check for pipeline.py with a pattern match for *pipeline\*.py.

Now it Supports non-standard modular pipeline structures.
```
- pipelines
  - sub_pipeline
    - pipeline_data_processing.py
    - sub_pipeline_1
        - pipeline_data_processing_1.py
```

### CODE REVIEW

**Review by noklam — COMMENTED** [review]
I think this looks good. Can you add some notes to https://github.com/kedro-org/vscode-kedro/blob/main/CHANGELOG.md.

It may be helpful to also update in README.md to explains what structure does the extension support.

Even though we don't have automated tests for this yet, I think we can also setup a sample project in this repo in `test` just for review purpose. It will be easier to write instructions for us what to test during review.

File: bundled/tool/lsp_server.py [hunk 1]
```diff
@@ -531,20 +532,22 @@ def references(
...
-        # Read the line number and match keywords naively
-        with open(pipeline_file) as f:
-            for i, line in enumerate(f):
-                if word in line:
-                    result.append((pipeline_file, i))
+        # Use glob to find files matching the pattern recursively
+        pipeline_files = glob.glob(f"{pipeline_dir}/**/*pipeline*.py", recursive=True)
+        for pipeline_file in pipeline_files:
+            if not os.path.exists(pipeline_file):
+                continue
```
Comment (noklam) [inline bundled/tool/lsp_server.py:544]: Is this still needed? What is the scenario that a filepath doesn't exist but showing up in `glob`?

Comment (jitu5) [reply bundled/tool/lsp_server.py:544]: There could be edge cases where a file is deleted or moved after `glob` has listed it but before the file is opened.

**Comment by jitu5** [comment]
> I think this looks good. Can you add some notes to https://github.com/kedro-org/vscode-kedro/blob/main/CHANGELOG.md.
> 
> It may be helpful to also update in README.md to explains what structure does the extension support.

I have updated README.md and CHANGELOG.md please review.

> Even though we don't have automated tests for this yet, I think we can also setup a sample project in this repo in `test` just for review purpose. It will be easier to write instructions for us what to test during review.

@noklam do you mean to add kedro project like `demo-project` in Kedro-Viz to `test` folder ? How can we run this extension over this sample kedro project ?

File: bundled/tool/lsp_server.py (hunk 1 above)
Comment (noklam) [reply bundled/tool/lsp_server.py:544]: This seems rather an edge case? I don't think we need to handle this specifically, since glob isn't taking that long and even if you do a check the file can still be deleted after the check but before it is opened.

Comment (jitu5) [reply bundled/tool/lsp_server.py:544]: Yes, agree its an edge case. let me remove this.

**Comment by noklam** [comment]
> @noklam do you mean to add kedro project like demo-project in Kedro-Viz to test folder ? How can we run this extension over this sample kedro project ?

Yes, I think it's good enough to have a test project, it will include multiple pipelines (maybe filename like pipeline_abc.py). We don't have to do this now. Let's discuss next time when we catch up.

Eventually I want to create some unit tests,[ similar to Kedro's e2e test setup.](https://github.com/kedro-org/kedro/tree/main/features/steps/test_starter/%7B%7B%20cookiecutter.repo_name%20%7D%7D)

**Review by noklam — APPROVED** [review]
